# LABORATORIO DIRIGIDO N.° 02
## Analítica Empresarial Integrada — TECSUP 2026-II

| | |
|---|---|
| **Docente** | Pilar Rocío Sayán Mejía |
| **Laboratorio** | Perfil analítico comparado de empresas del sector azucarero peruano |
| **N.° de laboratorio** | 2 |
| **Tema** | Estadios de evolución analítica y modelo DELTA de madurez organizacional |
| **Plan curricular** | Big Data y Ciencia de Datos · Código C28 · 2026-II |
| **Duración** | 190 minutos (7:00 p.m. – 10:10 p.m., con receso de 15 minutos) |
| **Fuente de datos** | SMV y BCRP — datos reales y oficiales |
| **Modalidad** |

---

**Apellidos y nombres del estudiante:** _Cuya Vera Juandiego Alejandro_

---


**Elemento de la capacidad terminal:** comprende los estadios de evolución analítica y las dimensiones del modelo DELTA y de su ampliación DELTA Plus; aplica estos marcos a la comparación de indicadores financieros reales y diferencia evidencia, inferencia y opinión.

> **Alcance.** Se analizan estados financieros individuales y anuales del ejercicio 2024. Los importes de la SMV se expresan en miles y pueden estar reportados en soles o dólares según la empresa. Las razones financieras son comparables cuando el numerador y el denominador pertenecen a la misma empresa, moneda y periodo.


## Actividad 1 — Revisión de conceptos (25 minutos)

**Propósito.** Consolidar el vocabulario técnico de la semana antes del desarrollo práctico. La precisión conceptual constituye un requisito del análisis: un indicador correctamente calculado pierde valor si se interpreta con categorías imprecisas.

**Instrucciones.** Complete la tabla con definiciones elaboradas con sus propias palabras. No se admite la reproducción literal de fuentes externas ni de sistemas generativos. La columna de la derecha contiene una pregunta de apoyo: si su definición permite responderla, la definición es suficiente; si no lo permite, corríjala antes de continuar.

**Evidencia esperada.** Tabla completa con las definiciones registradas.


| Concepto | Definición elaborada por el estudiante | Pregunta de apoyo |
|---|---|---|
| **Estadios de la evolución analítica** | Representan los distintos niveles por los que atraviesa una organización a medida que incorpora la analítica, desde un uso básico hasta convertirla en parte de sus decisiones y estrategia. | ¿Qué distingue a una organización del estadio 2 de una del estadio 3? |
| **Modelo DELTA** | Es un marco para analizar la capacidad analítica de una empresa mediante cinco componentes: Datos, Empresa, Liderazgo, Targets y Analistas. | ¿Por qué el modelo sostiene que las dimensiones deben avanzar de manera conjunta? |
| **Modelo DELTA Plus** | Es una extensión de DELTA que agrega factores adicionales para evaluar mejor si la analítica puede generar valor de forma continua dentro de la organización. | ¿Qué limitación del modelo original corrige la ampliación? |
| **Dimensión Technology** | Analiza si la empresa dispone de infraestructura y herramientas tecnológicas suficientes para almacenar, integrar, procesar y aprovechar sus datos. | ¿Puede una organización con datos y analistas presentar madurez limitada? Justifique. |
| **Dimensión Analytical Techniques** | Considera los métodos analíticos usados para convertir datos en información útil y responder diferentes preguntas del negocio. | ¿Qué diferencia una técnica descriptiva de una diagnóstica? Proponga un ejemplo. |
| **Organización que compite con analítica** | Es aquella que incorpora datos y análisis de manera sistemática en sus decisiones, procesos y estrategia para mejorar su desempeño frente a otras organizaciones. | ¿Basta con disponer de tableros de control para competir con analítica? |
| **Ventaja competitiva basada en datos** | Es un beneficio obtenido al usar los datos para decidir mejor y de una manera que no resulta sencilla de reproducir rápidamente por los competidores. | ¿Por qué razones esta ventaja puede resultar difícil de imitar por un competidor? |
| **Impacto de la IA generativa en la empresa** | La IA generativa puede acelerar tareas, apoyar el análisis y facilitar la interpretación de información, aunque las decisiones relevantes siguen requiriendo criterio y responsabilidad humana. | ¿Qué tarea del análisis financiero no debería delegarse a un sistema generativo? |
| **Perfil analítico comparado** | Consiste en contrastar las capacidades o resultados analíticos de varias empresas utilizando condiciones y criterios equivalentes. | ¿Qué condiciones deben cumplirse para que dos empresas sean comparables? |
| **Indicador financiero como evidencia** | Es una medida obtenida de información financiera que ayuda a evaluar una situación, siempre que la fuente sea confiable y el indicador corresponda con la pregunta que se desea analizar. | ¿En qué caso un indicador correctamente calculado no constituye evidencia válida? |


## Marco conceptual — Del modelo DELTA al modelo DELTA Plus

El modelo DELTA fue formulado por Davenport, Harris y Morison en *Analytics at Work* (2010) como un instrumento para evaluar la madurez analítica de una organización a partir de cinco dimensiones: **Data**, referida a la disponibilidad, integración y calidad de los datos; **Enterprise**, que examina si la capacidad analítica se encuentra dispersa en áreas aisladas o coordinada a escala institucional; **Leadership**, que valora el compromiso efectivo de la dirección con la decisión basada en evidencia; **Targets**, que analiza si los esfuerzos analíticos se concentran en los objetivos de mayor valor para el negocio; y **Analysts**, que considera la disponibilidad de profesionales con las competencias requeridas. El principio central del modelo establece que estas dimensiones deben progresar de manera conjunta, dado que un avance aislado en una de ellas no eleva la madurez global de la organización.

Con posterioridad, y ante la transformación del entorno tecnológico de la analítica empresarial, el modelo fue ampliado bajo la denominación **DELTA Plus**, que incorpora dos dimensiones adicionales. La primera es **Technology**, que comprende la infraestructura y las plataformas que permiten desarrollar y desplegar capacidades analíticas: soluciones en la nube, *data warehouses*, *data lakes* y arquitecturas *lakehouse*, entornos de procesamiento de grandes volúmenes de información, herramientas de visualización, plataformas analíticas corporativas, componentes de código abierto e infraestructura para el desarrollo de modelos de aprendizaje automático e inteligencia artificial. Su incorporación obedece a una constatación de orden práctico: una organización puede disponer de datos pertinentes y de analistas calificados y presentar, aun así, una madurez analítica limitada si carece de la tecnología necesaria para almacenar, procesar, analizar y poner en uso esa información de manera periódica y reproducible.

La segunda dimensión añadida es **Analytical Techniques**, que evalúa la variedad y la sofisticación de las técnicas empleadas para convertir los datos en información útil para la decisión. Estas técnicas admiten una progresión reconocible: la analítica **descriptiva** establece qué ocurrió; la **diagnóstica** indaga por qué ocurrió; la **predictiva** estima qué podría ocurrir; y la **prescriptiva** determina qué conviene hacer. Sobre esa base se incorporan progresivamente el aprendizaje automático, las técnicas de segmentación, los modelos de clasificación, los pronósticos, los métodos de optimización y las aplicaciones de inteligencia artificial. Dichas técnicas se presentan aquí como ilustración de la evolución de la capacidad analítica de una organización y serán objeto de desarrollo en semanas posteriores del curso; el presente laboratorio se sitúa en los niveles descriptivo y diagnóstico.

### Tabla comparativa

| | Modelo DELTA (2010) | Modelo DELTA Plus |
|---|---|---|
| **Dimensiones** | Cinco | Siete |
| **Componentes** | Data, Enterprise, Leadership, Targets, Analysts | Las cinco anteriores, más Technology y Analytical Techniques |
| **Objeto de evaluación** | Condiciones organizacionales de la analítica | Condiciones organizacionales y capacidad técnica de ejecución |
| **Pregunta que responde** | ¿Dispone la organización de datos, dirección, objetivos y personas? | Además de lo anterior, ¿cuenta con la tecnología y las técnicas para ejecutarlo de manera sostenida? |
| **Limitación que corrige** | — | Una organización con datos y analistas puede presentar madurez limitada por ausencia de tecnología o por el uso de técnicas elementales |

> **Precisión conceptual.** DELTA y DELTA Plus no constituyen modelos alternativos. DELTA Plus es una ampliación del modelo original: conserva íntegramente sus cinco dimensiones y añade dos. Al emitir un diagnóstico debe indicarse explícitamente cuál de los dos marcos se está aplicando.

---


## Actividad 2 — Desarrollo práctico y ejecución

**Propósito:** descargar información financiera anual y real de empresas del sector azucarero peruano, construir un perfil comparado de rentabilidad, liquidez y endeudamiento, y relacionarlo prudentemente con el contexto sectorial del BCRP mediante pandas, Polars, DuckDB y Plotly.

**Empresas:** Casa Grande S.A.A., Cartavio S.A.A., Agroindustrial Laredo S.A.A. y Empresa Agroindustrial Pomalca S.A.A.


## Paso 1: Preparación del entorno


In [1]:
# Ejecute esta celda en Google Colab. Las versiones se fijan para favorecer la reproducibilidad.
# %pip install permite instalar librerías desde el entorno de Google Colab.
%pip install -q polars==1.17.1 duckdb==1.1.3 econdata==1.0.17


# json permite trabajar con información en formato JSON.
# html permite trabajar y limpiar contenido relacionado con HTML.
# re permite utilizar expresiones regulares para buscar o modificar texto.
# unicodedata permite normalizar caracteres y texto, por ejemplo, tildes.
import json, html, re, unicodedata


# ElementTree permite leer y procesar información estructurada en formato XML.
from xml.etree import ElementTree as ET

# requests permite realizar solicitudes a páginas o servicios web para obtener datos.
import requests

# pandas permite trabajar con datos mediante DataFrames y realizar análisis.
import pandas as pd

# numpy proporciona herramientas para trabajar con números y cálculos.
import numpy as np

# polars es otra librería para procesar y analizar grandes cantidades de datos.
import polars as pl

# duckdb permite realizar consultas SQL directamente sobre los datos.
import duckdb

# plotly permite crear gráficos interactivos.
import plotly.graph_objects as go

# display permite mostrar tablas y otros objetos de manera más ordenada en Colab.
from IPython.display import display

# BCRP permite acceder a las series de información económica del Banco Central
# de Reserva del Perú.
from econdata import BCRP

# Configura el ancho de visualización de pandas para que las tablas se vean mejor.
pd.set_option("display.width", 160)

# Permite mostrar hasta 40 columnas de un DataFrame sin ocultarlas.
pd.set_option("display.max_columns", 40)

# Confirma que todas las librerías y configuraciones necesarias están listas.
print("Entorno listo.")

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 36.4/36.4 MB 31.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 20.2/20.2 MB 39.6 MB/s eta 0:00:00
Entorno listo.


> **Interpretación**:
Esta celda configura el entorno necesario para desarrollar el laboratorio. Se instalan y cargan las librerías que permitirán obtener, transformar, consultar y representar los datos financieros. Se utilizarán pandas y Polars para manipulación de datos, DuckDB para consultas SQL, Plotly para las visualizaciones y fuentes del BCRP para complementar el análisis con información económica oficial. Con ello se deja preparado el entorno antes de trabajar con las empresas del sector azucarero.

---


## Paso 2: Descarga de información financiera real desde la SMV

Se emplea el servicio web oficial de la SMV para consultar las principales cuentas y el Estado de Situación Financiera. Las operaciones utilizadas son `obtener_InfoFinanciera` y `obtener_BalanceGeneral`. El nombre oficial `obtener_GanciaPerdida` también se conserva para consultas del Estado de Resultados.


In [2]:
# Servicio web oficial de la SMV para obtener información financiera.
SERVICIO_SMV = "https://mvnet.smv.gob.pe/ws_od_eeff/WebServiceInfoFinanciera.asmx"

# Definimos el año, periodo y tipo de información que consultaremos.
EJERCICIO = 2024
PERIODO = "A"   # A = información anual
TIPO = "I"      # I = información individual

# Empresas azucareras que serán analizadas.
AZUCARARERAS = ["CASA GRANDE", "CARTAVIO", "AGROINDUSTRIAL LAREDO", "POMALCA"]

def etiqueta_local(etiqueta):
    # Retira el espacio de nombres de una etiqueta XML.
    return str(etiqueta).split("}")[-1]

def payload_a_dataframe(payload):
    # Convierte la respuesta JSON o XML de la SMV en un DataFrame.
    payload = html.unescape(str(payload)).strip()

    try:
        # Primero intenta interpretar la respuesta como JSON.
        objeto = json.loads(payload)

        # Busca los registros dentro de las claves habituales.
        if isinstance(objeto, dict):
            objeto = objeto.get("rows", objeto.get("data", objeto))

        # Convierte un único registro en una lista.
        if isinstance(objeto, dict):
            objeto = [objeto]

        # Finalmente convierte los datos en DataFrame.
        return pd.DataFrame(objeto)

    except json.JSONDecodeError:
        # Si no es JSON, interpreta la respuesta como XML.
        raiz = ET.fromstring(payload)
        registros = []

        # Recorre los nodos para encontrar registros con datos.
        for nodo in raiz.iter():
            hijos = list(nodo)
            if len(hijos) >= 5 and all(not list(hijo) for hijo in hijos):
                registros.append({
                    etiqueta_local(h.tag): h.text for h in hijos
                })

        return pd.DataFrame(registros)

def descargar_smv(operacion, ejercicio=EJERCICIO, periodo=PERIODO, tipo=TIPO):
    # Consulta una operación del servicio oficial de la SMV.

    # Construye la solicitud SOAP que se enviará a la SMV.
    sobre = f'''<?xml version="1.0" encoding="utf-8"?>
    <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
      <soap:Body><{operacion} xmlns="http://tempuri.org/">
        <Ejercicio>{ejercicio}</Ejercicio><Periodo>{periodo}</Periodo><Tipo>{tipo}</Tipo>
      </{operacion}></soap:Body>
    </soap:Envelope>'''

    # Envía la solicitud al servicio web.
    respuesta = requests.post(
        SERVICIO_SMV,
        data=sobre.encode("utf-8"),
        timeout=90,
        headers={
            "Content-Type": "text/xml; charset=utf-8",
            "SOAPAction": f'"http://tempuri.org/{operacion}"',
        },
    )

    # Detiene la ejecución si la SMV devuelve un error.
    respuesta.raise_for_status()

    # Interpreta la respuesta recibida como XML.
    raiz = ET.fromstring(respuesta.content)

    # Busca dentro de la respuesta los datos solicitados.
    resultado = next(
        (n for n in raiz.iter() if etiqueta_local(n.tag) == f"{operacion}Result"),
        None,
    )

    # Verifica que la SMV haya devuelto información.
    if resultado is None or not (resultado.text or "").strip():
        raise ValueError(f"La SMV no devolvió datos para {operacion}.")

    # Convierte los datos recibidos en DataFrame.
    datos = payload_a_dataframe(resultado.text)

    # Verifica que existan registros.
    if datos.empty:
        raise ValueError(f"La respuesta de {operacion} no contiene registros.")

    return datos

# Descarga las principales cuentas financieras.
principales = descargar_smv("obtener_InfoFinanciera")

# Descarga el estado de situación financiera.
balance = descargar_smv("obtener_BalanceGeneral")

# Muestra el tamaño de cada conjunto de datos.
print("Principales cuentas:", principales.shape)
print("Estado de situación financiera:", balance.shape)

# Muestra las primeras tres filas de las principales cuentas.
display(principales.head(3))

Principales cuentas: (276, 16)
Estado de situación financiera: (20574, 15)


,RPJ,TipoEmpresa,TipoSector,NombreEmpresa,RUC,CIIU,Ejercicio,TipoInformacion,Trimestre,Moneda,MetodoFlujoEfectivo,ActivoTotal,PatrimonioTotal,TotalIngreso,UtilidadNeta,PasivoTotal
0,L00474,EMPRESAS MERCADO ALTERNATIVO DE VALORES,DIVERSOS,A. JAIME ROJAS REPRESENTACIONES GENERALES S.A.,20102032951,5190,2024,Anual Individual,Anual,Soles,M todo Directo,77710,39497,77196,6786,38213
1,I00004,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION,,AC CAPITALES SOCIEDAD ADMINISTRADORA DE FONDOS...,20504893295,6430,2024,Anual Individual,Anual,Soles,M todo Directo,9070,7373,4880,-85,1697
2,OE7511,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION,,ACRES SOCIEDAD ADMINISTRADORA DE FONDOS DE INV...,20601498996,6430,2024,Anual Individual,Anual,Soles,M todo Indirecto,2724,2238,2348,176,486


> **Interpretación**:
En esta celda se consulta información financiera real correspondiente a 2024 desde la SMV para las empresas azucareras elegidas. Primero se definen los parámetros y funciones necesarias para realizar las solicitudes, procesar las respuestas y convertirlas en DataFrames. Después se descargan las principales cuentas financieras y el estado de situación financiera, comprobando que la información esté disponible antes de continuar.

---


## Paso 3: Normalización, filtrado y validación del sector

Se normalizan los nombres de las columnas, se filtran las cuatro empresas y se verifica que todas estén representadas. No se utiliza una base simulada ni un respaldo inventado.


In [4]:
# Normaliza los nombres de las columnas para facilitar su uso.
def nombre_columna(texto):
    texto = re.sub(r"(?<!^)(?=[A-Z])", "_", str(texto))  # Separa palabras unidas por mayúsculas.
    texto = unicodedata.normalize("NFKD", texto).encode("ascii", "ignore").decode()  # Quita tildes y caracteres especiales.
    return re.sub(r"[^a-z0-9]+", "_", texto.lower()).strip("_")  # Pasa a minúsculas y reemplaza símbolos.

# Prepara una copia del DataFrame y normaliza sus columnas.
def preparar(df):
    df = df.copy()  # Crea una copia para no modificar el original.
    df.columns = [nombre_columna(c) for c in df.columns]  # Aplica la normalización a cada columna.
    return df

# Busca una columna aunque tenga diferencias en su nombre.
def resolver_columna(df, candidatos):
    '''Encuentra una columna aun cuando la SMV cambie mayúsculas o convenciones de nombre.'''
    disponibles = {nombre_columna(c): c for c in df.columns}  # Normaliza los nombres disponibles.
    for candidato in candidatos:
        clave = nombre_columna(candidato)  # Normaliza el nombre que estamos buscando.
        if clave in disponibles:
            return disponibles[clave]  # Devuelve el nombre original de la columna encontrada.
    raise KeyError(f"No se encontró ninguna de estas columnas: {candidatos}. Disponibles: {list(df.columns)}")  # Informa si no encuentra ninguna.

# Estandariza la columna que identifica a la empresa.
def estandarizar_empresa(df):
    df = preparar(df)  # Normaliza primero las columnas.
    origen = resolver_columna(df, ["nombre_empresa", "razon_social", "empresa"])  # Busca la columna de empresa.
    return df.rename(columns={origen: "nombre_empresa"})  # Le asigna un nombre común.

# Empresas azucareras que serán analizadas.
AZUCARERAS = ["CASA GRANDE", "CARTAVIO", "AGROINDUSTRIAL LAREDO", "POMALCA"]

# Filtra únicamente las empresas azucareras seleccionadas.
def filtrar_azucareras(df):
    patron = "|".join(map(re.escape, AZUCARERAS))  # Crea un patrón con los nombres de las empresas.
    return df[
        df["nombre_empresa"].astype(str).str.upper().str.contains(patron, regex=True, na=False)
    ].copy()  # Conserva las filas que pertenecen a las empresas buscadas.

# Estandariza y filtra las dos fuentes obtenidas de la SMV.
principales = filtrar_azucareras(estandarizar_empresa(principales))
balance = filtrar_azucareras(estandarizar_empresa(balance))

# Obtiene los nombres únicos de las empresas encontradas.
empresas = sorted(principales["nombre_empresa"].dropna().unique())
print("Empresas encontradas:")
for empresa in empresas:
    print("-", empresa)  # Muestra cada empresa encontrada.

# Comprueba que las cuatro empresas solicitadas estén presentes.
faltantes = [nombre for nombre in AZUCARERAS if not any(nombre in e.upper() for e in empresas)]
if faltantes:
    raise ValueError(f"No se localizaron todas las empresas. Faltantes: {faltantes}")  # Detiene el proceso si falta alguna empresa.


Empresas encontradas:
- AGROINDUSTRIAL LAREDO S.A.A.
- CARTAVIO SOCIEDAD ANONIMA ABIERTA
- CASA GRANDE SOCIEDAD ANONIMA ABIERTA
- EMPRESA AGROINDUSTRIAL POMALCA S.A.A.


**Interpretación**:

Esta celda depura y comprueba los datos obtenidos de la SMV. Se uniformizan los nombres de las columnas, se identifica el campo correspondiente a la empresa y se filtran únicamente las cuatro compañías seleccionadas. Finalmente se valida que todas estén incluidas, con el fin de evitar que el análisis se realice sobre un conjunto incompleto.

---


### Paso 3: La cuenta correcta no se busca por su nombre

El Estado de Situación Financiera trae 69 cuentas por empresa, y **más de una contiene el texto «Activos Corrientes»**, pero solo una es el total. Filtrar por la descripción es un error frecuente y silencioso: no lanza ninguna excepción, devuelve un número, y ese número está mal.

Ejecute la celda siguiente y observe el catálogo: la cuenta `1D0122` —«Activos Corrientes Distintos al Efectivo Pignorados como Garantía Colateral»— vale 0 y, de usarse, produce una razón corriente igual a cero.


In [5]:
# Muestra las cuentas que contienen "corriente" para una empresa.
muestra = balance[balance["nombre_empresa"] == empresas[0]]  # Selecciona la primera empresa encontrada.

# Identifica la columna que contiene la descripción de la cuenta.
columna_desc = resolver_columna(muestra, ["descripcion_cuenta", "descripcion"])  # Busca el nombre correcto de la columna.

# Filtra las cuentas relacionadas con "corriente".
catalogo = muestra[
    muestra[columna_desc].astype(str).str.contains("orriente", na=False)  # Busca "corriente" en la descripción.
][["cuenta", columna_desc, "monto1"]]  # Conserva código, descripción y monto de la cuenta.

print("Empresa de referencia:", empresas[0])  # Muestra la empresa analizada.
display(catalogo.reset_index(drop=True))  # Muestra el catálogo de cuentas encontrado.

# Define los códigos oficiales de los totales que se usarán en los cálculos.
TOTAL_ACTIVO_CORRIENTE = "1D01ST"  # Código del total de activos corrientes.
TOTAL_PASIVO_CORRIENTE = "1D03ST"  # Código del total de pasivos corrientes.

print("Códigos declarados:", TOTAL_ACTIVO_CORRIENTE, TOTAL_PASIVO_CORRIENTE)  # Muestra ambos códigos.

Empresa de referencia: AGROINDUSTRIAL LAREDO S.A.A.


,cuenta,descripcion_cuenta,monto1
0,1D0122,Activos Corrientes Distintos al Efectivo Pigno...,0.0
1,1D0118,Total Activos Corrientes Distintos de los Acti...,137966.0
2,1D0115,Activos no Corrientes o Grupos de Activos para...,0.0
3,1D01ST,Total Activos Corrientes,137966.0
4,1D0223,"Activos por Impuestos Corrientes, no Corrientes",0.0
5,1D0224,Activos no Corrientes Distintos al Efectivo Pi...,0.0
6,1D02ST,Total Activos No Corrientes,802358.0
7,1D0315,Total de Pasivos Corrientes Distintos de Pasiv...,125303.0
8,1D03ST,Total Pasivos Corrientes,125303.0
9,1D0412,"Pasivos por Impuestos Corrientes, no Corriente",0.0


Códigos declarados: 1D01ST 1D03ST


> **Interpretación**:
Esta celda evidencia que buscar una cuenta solamente por su nombre puede llevar a seleccionar un registro incorrecto. Existen varias descripciones relacionadas con cuentas corrientes, pero no todas representan los totales requeridos. Por ese motivo, para calcular la razón corriente se emplean los códigos oficiales `1D01ST` y `1D03ST`, reduciendo el riesgo de usar una cuenta específica que distorsione el resultado.

---


## Paso 4: Construcción reproducible de los indicadores financieros

| Familia | Indicador | Fórmula |
|---|---|---|
| Rentabilidad | Margen neto | Utilidad neta / Ingresos |
| Rentabilidad | ROA | Utilidad neta / Activo total |
| Rentabilidad | ROE | Utilidad neta / Patrimonio |
| Liquidez | Razón corriente | Activo corriente / Pasivo corriente |
| Endeudamiento | Endeudamiento | Pasivo total / Activo total |
| Endeudamiento | Apalancamiento | Pasivo total / Patrimonio |

La razón corriente se obtiene del Estado de Situación Financiera mediante las descripciones oficiales de activos y pasivos corrientes.


In [6]:
# Renombra una columna usando el primer nombre disponible entre los candidatos.
def renombrar_primera(df, destino, candidatos):
    if destino in df.columns:  # Si ya tiene el nombre correcto, no hace cambios.
        return df
    return df.rename(columns={resolver_columna(df, candidatos): destino})  # Encuentra y renombra la columna.


# Estandariza los nombres de las principales cuentas financieras.
for destino, candidatos in {
    "ventas": ["total_ingreso", "total_ingresos", "ingresos", "ventas_netas"],
    "utilidad_neta": ["utilidad_neta", "ganancia_perdida_neta", "resultado_neto"],
    "activo_total": ["activo_total", "total_activo", "total_activos"],
    "pasivo_total": ["pasivo_total", "total_pasivo", "total_pasivos"],
    "patrimonio": ["patrimonio_total", "total_patrimonio", "patrimonio"],
}.items():
    principales = renombrar_primera(principales, destino, candidatos)  # Aplica el nombre estándar.


# Selecciona las cuentas necesarias para calcular los indicadores.
columnas_base = [
    "nombre_empresa", "ventas", "utilidad_neta", "activo_total",
    "pasivo_total", "patrimonio"
]

# Elimina registros repetidos y crea una copia para el perfil.
perfil = principales[columnas_base].drop_duplicates("nombre_empresa").copy()

# Cambia el nombre de la empresa para facilitar el análisis.
perfil = perfil.rename(columns={"nombre_empresa": "empresa"})


# Convierte los valores financieros de la SMV a números.
def a_numero(valor):
    '''Convierte importes de la SMV aunque incluyan separadores, espacios o paréntesis.'''
    if pd.isna(valor):  # Si el valor está vacío, devuelve un valor nulo.
        return np.nan

    if isinstance(valor, (int, float, np.number)):  # Si ya es numérico, lo convierte a float.
        return float(valor)

    texto = str(valor).strip().replace("\u00a0", "").replace(" ", "")  # Limpia espacios.

    negativo = texto.startswith("(") and texto.endswith(")")  # Detecta valores negativos entre paréntesis.
    texto = texto.strip("()")  # Elimina los paréntesis.

    texto = texto.replace(",", "")  # Elimina separadores de miles.
    texto = re.sub(r"[^0-9.\-]", "", texto)  # Conserva solo caracteres numéricos.

    if texto in {"", "-", "."}:  # Identifica valores que no pueden convertirse.
        return np.nan

    numero = float(texto)  # Convierte el texto a número.
    return -numero if negativo else numero  # Devuelve el número con su signo correcto.


# Convierte a números todas las cuentas que se utilizarán en los cálculos.
for columna in ["ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio"]:
    perfil[columna] = perfil[columna].map(a_numero)


# Extrae los totales corrientes utilizando sus códigos oficiales.
def extraer_total_corriente(df, codigo, nombre):
    col_codigo = resolver_columna(df, ["cuenta", "codigo_cuenta", "codigo"])  # Identifica la columna del código.
    monto = resolver_columna(df, ["monto1", "monto", "importe", "valor"])  # Identifica la columna del monto.

    datos = df[df[col_codigo].astype(str).str.strip() == codigo][
        ["nombre_empresa", monto]].copy()  # Selecciona únicamente el código solicitado.

    datos[monto] = datos[monto].map(a_numero)  # Convierte los montos a números.
    datos = datos.dropna().drop_duplicates("nombre_empresa")  # Elimina vacíos y duplicados.

    if len(datos) < len(AZUCARERAS):  # Comprueba que estén las cuatro empresas.
        raise ValueError(
            f"La cuenta {codigo} no está disponible para las cuatro empresas "
            f"en el ejercicio {EJERCICIO}."
        )

    return datos.rename(columns={"nombre_empresa": "empresa", monto: nombre})  # Devuelve los datos estandarizados.


# Agrega el activo corriente al perfil.
perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_ACTIVO_CORRIENTE, "activo_corriente"),
    on="empresa", how="inner")

# Agrega el pasivo corriente al perfil.
perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_PASIVO_CORRIENTE, "pasivo_corriente"),
    on="empresa", how="inner")


# Verifica que estén las cuatro empresas.
if len(perfil) < 4:
    raise ValueError("No fue posible construir el perfil completo de las cuatro empresas.")


# Define las cuentas que deben tener valores para calcular los indicadores.
columnas_calculo = [
    "ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio",
    "activo_corriente", "pasivo_corriente"
]

# Busca valores faltantes antes de realizar los cálculos.
faltantes = perfil[columnas_calculo].isna().sum()

if faltantes.any():
    raise ValueError(
        "Existen cuentas sin valor numérico: " +
        ", ".join(f"{c} ({n})" for c, n in faltantes.items() if n)
    )


# Verifica que los principales denominadores no sean cero.
if (perfil[["ventas", "activo_total", "patrimonio", "pasivo_corriente"]] == 0).any().any():
    raise ValueError("Existe un denominador igual a cero; revise las cuentas antes de calcular.")


# Calcula los indicadores de rentabilidad.
perfil["margen_neto"] = perfil["utilidad_neta"] / perfil["ventas"]
perfil["roa"] = perfil["utilidad_neta"] / perfil["activo_total"]
perfil["roe"] = perfil["utilidad_neta"] / perfil["patrimonio"]

# Calcula el indicador de liquidez.
perfil["razon_corriente"] = perfil["activo_corriente"] / perfil["pasivo_corriente"]

# Calcula los indicadores de endeudamiento.
perfil["endeudamiento"] = perfil["pasivo_total"] / perfil["activo_total"]
perfil["apalancamiento"] = perfil["pasivo_total"] / perfil["patrimonio"]


# Comprueba que los cálculos no hayan generado infinitos.
numeros = perfil.select_dtypes(include="number").to_numpy()

if np.isinf(numeros[~np.isnan(numeros)]).any():
    raise ValueError("Se obtuvo una división entre cero; revise las cuentas de la empresa afectada.")


# Muestra el perfil financiero de las cuatro empresas.
display(perfil.round(3))

,empresa,ventas,utilidad_neta,activo_total,pasivo_total,patrimonio,activo_corriente,pasivo_corriente,margen_neto,roa,roe,razon_corriente,endeudamiento,apalancamiento
0,AGROINDUSTRIAL LAREDO S.A.A.,438934.0,55453.0,940324.0,327342.0,612982.0,137966.0,125303.0,0.126,0.059,0.090,1.101,0.348,0.534
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,535074.0,91846.0,926161.0,542687.0,383474.0,294933.0,307537.0,0.172,0.099,0.240,0.959,0.586,1.415
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,637681.0,73490.0,1923732.0,874083.0,1049649.0,429304.0,362479.0,0.115,0.038,0.070,1.184,0.454,0.833
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,234045.0,9636.0,1352469.0,679594.0,672875.0,168835.0,108618.0,0.041,0.007,0.014,1.554,0.502,1.010


> **Interpretación**:
Esta celda genera un perfil financiero comparable para las cuatro empresas utilizando información oficial de la SMV. Se revisan y estandarizan las cuentas, se incorporan los valores de activos y pasivos corrientes mediante sus códigos oficiales y luego se calculan indicadores de rentabilidad, liquidez y endeudamiento. Las comprobaciones incluidas ayudan a prevenir resultados erróneos por datos faltantes o divisiones entre cero.


**Pregunta 1.** Identifique la empresa con mayor ROE y la empresa con mayor endeudamiento. ¿Coinciden? Interprete la relación observada sin afirmar causalidad.

**Respuesta:**

---


### Reto 1 — Predicción antes del ranking (5 minutos)

**Sin ordenar todavía la tabla**, responda en equipo:

1. ¿Qué empresa creen que presenta el mejor perfil financiero?
2. Sustenten la predicción con **dos indicadores** como mínimo.
3. ¿Qué indicador podría hacer cambiar su conclusión?

**Predicción del equipo:**
Se considera que **CARTAVIO SOCIEDAD ANONIMA ABIERTA** podría presentar el mejor perfil financiero.

- **Justificación:**
La elección se apoya principalmente en su **ROE de 0.240**, que es el mayor del grupo, y en su **margen neto de 0.172**, también ubicado en el primer lugar. A partir de estos dos indicadores, Cartavio muestra un nivel de rentabilidad superior al de las demás empresas analizadas.

- **Indicador que podría hacer cambiar nuestra conclusión:**  
El **endeudamiento** podría modificar esta apreciación, debido a que Cartavio registra **0.586**, el valor más alto del conjunto. También debe considerarse su **razón corriente de 0.959**, situada por debajo de 1.

Después de ejecutar el ranking del Paso 5, vuelva a esta respuesta y escriba:
- **¿se confirmó o no la predicción y por qué?**

La predicción **sí se confirmó**, ya que **CARTAVIO SOCIEDAD ANONIMA ABIERTA** alcanzó el mayor **ROE (0.240)** y el mayor **margen neto (0.172)**. No obstante, estos resultados de rentabilidad aparecen acompañados por el **endeudamiento más alto (0.586)** y una razón corriente de **0.959**, por lo que su situación debe evaluarse considerando simultáneamente rentabilidad y obligaciones financieras.

---


## Paso 5: Comparación sectorial con Polars y DuckDB


In [ ]:
perfil_pl = pl.from_pandas(perfil)

ranking_roe = (
    perfil_pl
    .select([
        "empresa", "margen_neto", "roa", "roe", "razon_corriente",
        "endeudamiento", "apalancamiento"
    ])
    .sort("roe", descending=True)
)
ranking_roe


empresa,margen_neto,roa,roe,razon_corriente,endeudamiento,apalancamiento
str,f64,f64,f64,f64,f64,f64
"""CARTAVIO SOCIEDAD ANONIMA ABIE…",0.171651,0.099169,0.23951,0.959016,0.585953,1.415186
"""AGROINDUSTRIAL LAREDO S.A.A.""",0.126336,0.058972,0.090464,1.101059,0.348116,0.534016
"""CASA GRANDE SOCIEDAD ANONIMA A…",0.115246,0.038202,0.070014,1.184356,0.454368,0.832738
"""EMPRESA AGROINDUSTRIAL POMALCA…",0.041172,0.007125,0.014321,1.554392,0.502484,1.009986


In [ ]:
# Misma comparación del ranking anterior, ahora expresada en SQL sobre DuckDB.
# Ejecutar la misma pregunta en dos motores distintos y obtener el mismo
# resultado es una comprobación de reproducibilidad, no una coincidencia.
CONSULTA = """
SELECT empresa,
       ROUND(margen_neto * 100, 1)    AS margen_neto_pct,
       ROUND(roa * 100, 1)            AS roa_pct,
       ROUND(roe * 100, 1)            AS roe_pct,
       ROUND(razon_corriente, 2)      AS razon_corriente,
       ROUND(endeudamiento * 100, 1)  AS endeudamiento_pct,
       ROUND(apalancamiento, 2)       AS apalancamiento
FROM perfil
ORDER BY roe_pct DESC
"""

consulta = duckdb.sql(CONSULTA).df()
display(consulta)

# Comprobación automática: DuckDB y Polars deben coincidir.
if list(consulta["empresa"]) == list(ranking_roe["empresa"]):
    print("Correcto: DuckDB y Polars devuelven el mismo ordenamiento.")
else:
    print("Los ordenamientos no coinciden. Revise las columnas y el ORDER BY.")


,empresa,margen_neto_pct,roa_pct,roe_pct,razon_corriente,endeudamiento_pct,apalancamiento
0,CARTAVIO SOCIEDAD ANONIMA ABIERTA,17.2,9.9,24.0,0.96,58.6,1.42
1,AGROINDUSTRIAL LAREDO S.A.A.,12.6,5.9,9.0,1.10,34.8,0.53
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,11.5,3.8,7.0,1.18,45.4,0.83
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4.1,0.7,1.4,1.55,50.2,1.01


Correcto: DuckDB y Polars devuelven el mismo ordenamiento.


In [ ]:
# Rankings complementarios para comparar dimensiones distintas del perfil.
ranking_comparado = pd.DataFrame({
    "empresa": perfil["empresa"],
    "rank_roe": perfil["roe"].rank(ascending=False, method="min").astype(int),
    "rank_liquidez": perfil["razon_corriente"].rank(ascending=False, method="min").astype(int),
    # En endeudamiento, un valor menor recibe mejor posición solo como criterio descriptivo de exposición financiera.
    "rank_endeudamiento": perfil["endeudamiento"].rank(ascending=True, method="min").astype(int),
})
ranking_comparado["cambio_max_posicion"] = (
    ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].max(axis=1)
    - ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].min(axis=1)
)
display(ranking_comparado.sort_values(["rank_roe", "empresa"]))


,empresa,rank_roe,rank_liquidez,rank_endeudamiento,cambio_max_posicion
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,1,4,4,3
0,AGROINDUSTRIAL LAREDO S.A.A.,2,3,1,2
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,3,2,2,1
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4,1,3,3


### Reto 2 — Evidencia, inferencia y conclusión (5 minutos)

Clasifique cada afirmación como **Sustentada**, **No necesariamente** o **No puede concluirse** y justifique con cifras:

| Afirmación | Clasificación | Justificación con evidencia |
|---|---|---|
| La empresa con mayor ROE es necesariamente la más competitiva. |  |  |
| Una empresa con mayor endeudamiento posee necesariamente menor madurez analítica. |  |  |
| Obtener un ordenamiento consistente con Polars y DuckDB fortalece la reproducibilidad del análisis. |  |  |

**Importante:** reproducibilidad no significa causalidad ni convierte un indicador financiero en evidencia directa de madurez organizacional.


**Pregunta 2.** Compare los rankings por rentabilidad, liquidez y endeudamiento. Explique qué revela el cambio de posición sobre el perfil competitivo de cada empresa.

**Respuesta:**

Los resultados evidencian que la posición de cada empresa depende del indicador que se tome como referencia. **Cartavio** lidera la rentabilidad con un ROE de **24.0 %**, pero se ubica en el cuarto puesto tanto en liquidez (**0.96**) como en endeudamiento, debido a que presenta el nivel más alto del grupo (**58.6 %**). Por ello, combina una rentabilidad elevada con una mayor exposición financiera.

**Agroindustrial Laredo** se sitúa segunda en ROE (**9.0 %**), tercera en liquidez (**1.10**) y primera al ordenar por menor endeudamiento (**34.8 %**). Estos valores muestran una situación relativamente balanceada entre rentabilidad, liquidez y exposición financiera.

**Casa Grande** ocupa el tercer lugar en rentabilidad, con un ROE de **7.0 %**, pero mejora al segundo puesto en liquidez (**1.18**) y también en endeudamiento (**45.4 %**). Esto indica que, aunque su rentabilidad es menor, su posición mejora cuando se consideran otras dimensiones financieras.

Por último, **Pomalca** queda cuarta en rentabilidad, con un ROE de **1.4 %**, pero alcanza el primer puesto en liquidez (**1.55**) y el tercero en endeudamiento (**50.2 %**). Esto confirma que sobresalir en una dimensión no garantiza obtener la misma posición en las demás.

En conjunto, los cambios observados entre rankings reflejan que el perfil competitivo debe evaluarse de manera **multidimensional**, integrando rentabilidad, liquidez y endeudamiento en lugar de analizar un solo indicador.

---


## Paso 6: Visualización del perfil comparado


In [ ]:
indicadores_pct = ["margen_neto", "roa", "roe", "endeudamiento"]
nombres_indicadores = {
    "margen_neto": "Margen neto",
    "roa": "ROA",
    "roe": "ROE",
    "endeudamiento": "Endeudamiento",
}
perfil_grafico = perfil.copy()
perfil_grafico["empresa_corta"] = (
    perfil_grafico["empresa"].str.upper()
    .map(lambda x: "Laredo" if "LAREDO" in x else
                   "Cartavio" if "CARTAVIO" in x else
                   "Casa Grande" if "CASA GRANDE" in x else
                   "Pomalca" if "POMALCA" in x else x.title())
)
fig = go.Figure()
for indicador in indicadores_pct:
    fig.add_trace(go.Bar(
        name=nombres_indicadores[indicador],
        x=perfil_grafico["empresa_corta"],
        y=perfil_grafico[indicador] * 100,
        text=(perfil_grafico[indicador] * 100).round(1),
        texttemplate="%{text:.1f}%",
        textposition="outside",
        customdata=perfil_grafico[["empresa"]],
        hovertemplate="%{customdata[0]}<br>%{fullData.name}: %{y:.2f}%<extra></extra>",
    ))
fig.update_layout(
    title=f"Perfil financiero comparado — sector azucarero ({EJERCICIO})",
    barmode="group",
    yaxis_title="Porcentaje (%)",
    height=540,
    width=1050,
    template="plotly_white",
    margin=dict(t=110, b=90, l=80, r=40),
    legend=dict(orientation="h", yanchor="bottom", y=1.02, xanchor="left", x=0),
)
fig.show()

fig_liquidez = go.Figure(go.Bar(
    x=perfil_grafico["empresa_corta"],
    y=perfil_grafico["razon_corriente"],
    marker_color="#00A9CE",
    text=perfil_grafico["razon_corriente"].round(2),
    texttemplate="%{text:.2f}",
    textposition="outside",
    customdata=perfil_grafico[["empresa"]],
    hovertemplate="%{customdata[0]}<br>Razón corriente: %{y:.2f}<extra></extra>",
))
fig_liquidez.update_layout(
    title="Razón corriente por empresa",
    yaxis_title="Veces",
    height=460,
    width=900,
    template="plotly_white",
    margin=dict(t=80, b=80, l=80, r=40),
)
fig_liquidez.update_yaxes(rangemode="tozero")
fig_liquidez.show()


## Paso 7: Contexto sectorial con datos específicos del BCRP

Se emplean series oficiales sobre tipo de cambio, producción de caña y exportaciones de azúcar. Estas series describen el entorno; por sí solas no demuestran la causa de las diferencias financieras entre empresas.


In [ ]:
SERIES = {
    "PN01209PM": "tipo_cambio",
    "PN01790AM": "produccion_cana_azucar_miles_t",
    "PN38774BM": "export_azucar_fob_millones_usd",
    "PN38775BM": "volumen_exportado_azucar",
}

contexto = BCRP.get_data(SERIES, "2020-01", "2024-12")
print("Datos del BCRP:", contexto.shape)
display(contexto.tail(12))


Datos del BCRP: (60, 4)


,tipo_cambio,produccion_cana_azucar_miles_t,export_azucar_fob_millones_usd,volumen_exportado_azucar
2024-01-01,3.743667,755.602441,7.547009,10.321329
2024-02-01,3.830905,814.525062,4.398561,5.993531
2024-03-01,3.712947,563.626363,1.911468,2.355260
2024-04-01,3.718409,634.675782,3.839971,4.951844
2024-05-01,3.735273,812.740966,3.235732,3.843619
2024-06-01,3.789526,830.637431,0.481814,0.618096
2024-07-01,3.769500,840.996455,0.919740,1.246593
2024-08-01,3.745550,1017.287110,1.010892,1.358606
2024-09-01,3.772143,930.472111,3.097093,4.088378
2024-10-01,3.757500,1053.902968,1.906427,2.543543


In [ ]:
# Evolución temporal del contexto sectorial. Cada serie se normaliza a índice 100
# para comparar tendencias con unidades diferentes, sin alterar los datos originales.
contexto_grafico = contexto.copy()
columnas_numericas = contexto_grafico.select_dtypes(include="number").columns
base = contexto_grafico[columnas_numericas].apply(lambda s: s.dropna().iloc[0])
indice_100 = contexto_grafico[columnas_numericas].divide(base).multiply(100)

fig_contexto = go.Figure()
for columna in indice_100.columns:
    fig_contexto.add_trace(go.Scatter(
        x=indice_100.index,
        y=indice_100[columna],
        mode="lines",
        name=columna,
    ))
fig_contexto.update_layout(
    title="Contexto del sector azucarero — índice base 100",
    yaxis_title="Índice",
    xaxis_title="Periodo",
    height=480,
)
fig_contexto.show()


**Pregunta 3.** Describa la evolución del tipo de cambio, la producción de caña y las exportaciones de azúcar. Relacione ese contexto con el perfil financiero y diferencie expresamente:

- **Evidencia:** resultado directamente observado.
- **Inferencia:** interpretación razonada basada en la evidencia.
- **Opinión:** valoración que debe justificarse.

**Respuesta:**

**Evidencia:**  
Durante el periodo 2020-2024, el **tipo de cambio** registra variaciones, aunque se mantiene alrededor de valores superiores a los observados al inicio del intervalo. La **producción de caña de azúcar** presenta también oscilaciones mensuales y alcanza niveles altos en varios meses de 2024. En cuanto a las **exportaciones de azúcar**, su comportamiento es irregular: muestran valores mayores al comenzar 2024 y luego disminuyen durante varios meses. Un ejemplo es el valor FOB exportado, que desciende de **7.55 millones de US$ en enero de 2024 a 0.48 millones de US$ en junio de 2024**.

**Inferencia:**  
Los datos muestran que las empresas azucareras operan en un entorno donde cambian la producción, el tipo de cambio y las exportaciones. Estas variables permiten dar contexto a los resultados financieros, pero la evidencia disponible **no demuestra que sean la causa** de las diferencias entre compañías. Por ejemplo, Cartavio registra el mayor ROE (**24.0 %**) junto con el mayor endeudamiento (**58.6 %**), mientras Pomalca posee el menor ROE (**1.4 %**) y la razón corriente más alta (**1.55**). Por ello, los indicadores deben interpretarse de manera conjunta.

**Opinión:**  
Considero que el contexto del sector aporta información útil para interpretar el desempeño empresarial, pero **no debería emplearse por sí solo para explicar la situación financiera** de cada compañía. Para establecer relaciones más sólidas sería necesario incorporar datos adicionales de las empresas y contrastarlos con la evolución de las variables sectoriales.

---


### Reto 3 — Diagnóstico provisional de madurez y DELTA (8 minutos)

Seleccione **una** de las cuatro empresas. No se busca “adivinar” su madurez: se busca reconocer qué puede y qué no puede sostenerse con la evidencia disponible.


| Aspecto | Respuesta |
|---|---|
| **Empresa seleccionada** | **CARTAVIO SOCIEDAD ANONIMA ABIERTA** |
| **Evidencia financiera observada** | Registra el mayor **ROE (24.0 %)** y el mayor **margen neto (17.2 %)**; al mismo tiempo, presenta el endeudamiento más elevado (**58.6 %**) y una razón corriente de **0.96**. |
| **Estadio de evolución analítica compatible de manera provisional** | Con los estados financieros disponibles **no es posible establecerlo de forma concluyente**. Estos resultados pueden servir únicamente para formular una hipótesis inicial, pero no ofrecen evidencia suficiente para asignar un estadio de madurez analítica. |
| **Dimensión DELTA evaluable parcialmente** | **Targets**, debido a que los indicadores permiten observar resultados vinculados con objetivos financieros, aunque no muestran si la empresa ha definido objetivos analíticos de manera formal. |
| **Dimensiones DELTA que no pueden verificarse** | **Data, Enterprise, Leadership y Analysts**, ya que los estados financieros no permiten conocer la calidad e integración de los datos, la coordinación de la analítica entre áreas, el liderazgo orientado por evidencia ni las capacidades de los especialistas. |
| **Evidencia organizacional adicional requerida** | Se necesitaría información sobre gobierno y calidad de datos, procesos de análisis, participación de la gerencia en decisiones sustentadas en datos, organización de los equipos analíticos, competencias de sus integrantes y uso de tecnologías analíticas e IA. |

**Cierre del reto:**  

**Evidencia:** Cartavio registra un ROE de **24.0 %** y un endeudamiento de **58.6 %**.

**Inferencia:** sus indicadores reflejan una combinación de alta rentabilidad y mayor exposición financiera.

**Opinión:** los resultados hacen conveniente revisar rentabilidad y endeudamiento de forma conjunta, pero no son suficientes para definir el nivel de madurez analítica de la empresa.


### Pausa metodológica — ¿qué NO podemos afirmar?

Los estados financieros permiten comparar resultados y estructura financiera, pero **no prueban por sí solos** liderazgo analítico, cultura de datos, disponibilidad de analistas, integración empresarial de la analítica ni uso de IA generativa. Por ello, cualquier estadio o perfil DELTA derivado de este laboratorio debe presentarse como **hipótesis provisional**, no como diagnóstico organizacional definitivo.


## Actividad 3 — Informe de perfil analítico comparado

Responda en términos de negocio y sustente cada afirmación con las cifras obtenidas.

> **A.** Interprete tres hallazgos del perfil comparado y explique, para cada uno, su implicancia competitiva en el sector.

**Respuesta:**

**1. Cartavio lidera los indicadores de rentabilidad.**  
Presenta un **ROE de 24.0 %** y un **margen neto de 17.2 %**, los valores más altos entre las cuatro empresas. Esto la coloca en una posición destacada desde la perspectiva de rentabilidad, aunque el resultado debe revisarse junto con su nivel de endeudamiento.

**2. Cartavio registra también el endeudamiento más elevado.**  
Su endeudamiento es de **58.6 %**, frente al **34.8 %** de Laredo, **45.4 %** de Casa Grande y **50.2 %** de Pomalca. Esto implica una exposición financiera superior dentro del grupo, por lo que su rentabilidad no debería evaluarse de manera aislada.

**3. Pomalca combina la menor rentabilidad con la mayor liquidez.**  
Su ROE es de solo **1.4 %**, el más bajo del conjunto, mientras que su razón corriente alcanza **1.55**, la cifra más alta. Esto demuestra que una mejor posición de liquidez no necesariamente se traduce en mayor rentabilidad.


> **B.** Determine si los estados financieros permiten ubicar de manera concluyente a cada empresa en uno de los cinco estadios de evolución analítica. Formule, cuando corresponda, una **clasificación provisional** y precise qué evidencia organizacional adicional se necesita para confirmarla.

**Respuesta:**

No. A partir de los estados financieros **no se puede asignar de forma concluyente** a las cuatro empresas uno de los cinco estadios de evolución analítica. Los indicadores calculados permiten comparar resultados financieros, pero no revelan de qué manera la analítica es utilizada dentro de cada organización.

Por esa razón, no corresponde establecer un estadio definitivo para Casa Grande, Cartavio, Agroindustrial Laredo o Pomalca. Como aproximación provisional, todas pueden mantenerse bajo la categoría **“estadio no determinado con la evidencia disponible”**.

Para confirmar una clasificación se necesitarían datos sobre gobierno y uso de la información, integración entre áreas, liderazgo, objetivos analíticos, competencias de los analistas, procesos de decisión y nivel de adopción tecnológica.


> **C.** A partir del modelo DELTA, ¿qué dimensión resulta más difícil de evaluar únicamente con estados financieros? Explique por qué y qué información adicional se requeriría.

**Respuesta:**

La dimensión **Leadership** es una de las más difíciles de analizar utilizando únicamente estados financieros, porque estos documentos muestran resultados económicos, pero no evidencian si la alta dirección impulsa, solicita o emplea decisiones sustentadas en datos.

Para evaluarla sería necesario conocer cómo participa la gerencia en los procesos analíticos, qué decisiones se toman con evidencia, qué indicadores revisan los directivos y cómo realizan el seguimiento de los resultados.

> **D.** Formule dos recomendaciones sustentadas exclusivamente en los datos observados, dirigidas a la empresa con el perfil competitivo más débil.

**Respuesta:**

La empresa con el perfil financiero más débil es **EMPRESA AGROINDUSTRIAL POMALCA S.A.A.**, ya que presenta el menor margen neto (**4.1 %**), ROA (**0.7 %**) y ROE (**1.4 %**) entre las compañías analizadas.

**1. Examinar la rentabilidad de sus operaciones**, revisando los elementos asociados con el margen neto de **4.1 %** y el ROA de **0.7 %** para identificar dónde se concentra el menor rendimiento.

**2. Revisar el aprovechamiento de sus activos y recursos financieros**, buscando posibilidades de elevar el retorno generado por los activos y el patrimonio sin atribuir el desempeño a causas que no estén demostradas por los datos.


> **E.** Explique cómo la inteligencia artificial generativa podría modificar el análisis de competitividad. Distinga entre usos plausibles y afirmaciones no verificables.

**Respuesta:**

La inteligencia artificial generativa podría aportar al análisis competitivo al acelerar la revisión de grandes volúmenes de información, resumir resultados, contrastar indicadores, reconocer patrones y colaborar en la construcción de escenarios para apoyar decisiones.

Entre sus aplicaciones plausibles estarían la automatización de consultas financieras, la generación de explicaciones sobre cambios en los indicadores y el apoyo en la evaluación de distintos escenarios.

Sin embargo, los estados financieros no permiten comprobar que alguna de las empresas analizadas utilice IA generativa ni que esta tecnología esté mejorando actualmente su competitividad o su madurez analítica. Para afirmarlo se requeriría evidencia sobre herramientas implementadas, procesos de adopción y resultados obtenidos.


### Matriz de síntesis para la decisión

Complete esta matriz antes de redactar las conclusiones finales:

| Empresa | Fortaleza principal | Riesgo/debilidad principal | Evidencia numérica | Implicancia competitiva |
|---|---|---|---|---|
| **Casa Grande** | Liquidez | Rentabilidad inferior a Cartavio y Laredo | ROE **7.0 %**; razón corriente **1.18** | Mantiene una liquidez favorable, aunque su rentabilidad se encuentra por debajo de las empresas que lideran ese indicador. |
| **Cartavio** | Rentabilidad | Endeudamiento más alto y razón corriente menor que 1 | ROE **24.0 %**; endeudamiento **58.6 %**; razón corriente **0.96** | Sobresale por su rentabilidad, pero al mismo tiempo muestra una mayor exposición financiera. |
| **Agroindustrial Laredo** | Menor endeudamiento | Liquidez en un nivel intermedio | ROE **9.0 %**; endeudamiento **34.8 %** | Muestra una combinación relativamente balanceada entre rentabilidad y nivel de exposición financiera. |
| **Pomalca** | Liquidez | Rentabilidad más baja | ROE **1.4 %**; ROA **0.7 %**; razón corriente **1.55** | Su buena liquidez no se refleja en un nivel de rentabilidad similar al de las otras empresas. |

**¿Qué empresa requiere mayor atención gerencial según los indicadores observados y qué dos acciones recomendaría sin exceder la evidencia disponible?**

La empresa que requiere mayor atención es **EMPRESA AGROINDUSTRIAL POMALCA S.A.A.**, porque presenta los resultados de rentabilidad más bajos del grupo: **ROE de 1.4 %, ROA de 0.7 % y margen neto de 4.1 %**.

**Acciones recomendadas:**

1. **Revisar los elementos asociados con su baja rentabilidad**, analizando el margen neto y los rendimientos generados sobre activos y patrimonio.

2. **Examinar cómo se están utilizando los activos y recursos disponibles**, considerando que la empresa tiene la mayor razón corriente (**1.55**) pero el menor ROE (**1.4 %**), con el objetivo de encontrar oportunidades para mejorar su rendimiento financiero.

Las recomendaciones se sustentan únicamente en los indicadores observados y no atribuyen los resultados a causas que no hayan sido verificadas.


---

## Reto de aplicación y retroalimentación

En esta sección se aplicarán los procedimientos desarrollados durante la sesión a nuevas situaciones de análisis. Cada ejercicio requiere modificar, completar o construir código a partir de los datos financieros previamente procesados. Posteriormente, los resultados obtenidos deberán interpretarse brevemente desde una perspectiva empresarial. El propósito es comprobar la comprensión de las técnicas utilizadas y fortalecer la capacidad de adaptar el análisis ante nuevas preguntas de negocio.

**Indicaciones generales.** Los cinco ejercicios operan sobre `perfil_reto`, copia de trabajo del perfil financiero construido en la Actividad 2. Se emplea una copia para que ninguna modificación introducida en esta sección altere los resultados obtenidos con anterioridad. Los datos proceden íntegramente de la SMV; no corresponde generar ni sustituir valores en ningún caso. Cada respuesta escrita no debe exceder cuatro líneas.

**Duración en sesión:** 25 minutos. Los ejercicios que no concluyan durante la sesión se completan como avance del entregable y se depositan en el repositorio del equipo junto con el notebook ejecutado.


In [9]:
# Copia de trabajo para la sección de retos.
perfil_reto = perfil.copy()
print("Copia de trabajo creada:", perfil_reto.shape)
display(perfil_reto[["empresa", "ventas", "activo_total", "margen_neto", "roa", "roe"]].round(3))


Copia de trabajo creada: (4, 14)


,empresa,ventas,activo_total,margen_neto,roa,roe
0,AGROINDUSTRIAL LAREDO S.A.A.,438934.0,940324.0,0.126,0.059,0.090
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,535074.0,926161.0,0.172,0.099,0.240
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,637681.0,1923732.0,0.115,0.038,0.070
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,234045.0,1352469.0,0.041,0.007,0.014


### Ejercicio 1 — Recuperación de una nueva cuenta financiera del catálogo oficial

Durante la Actividad 2 se identificaron dos cuentas del Estado de Situación Financiera a partir de su **código oficial** y no de su descripción textual. Ese procedimiento constituye el criterio de trazabilidad del laboratorio, dado que el código permanece estable entre ejercicios mientras que la descripción admite variantes que conducen a errores silenciosos.

En este ejercicio deberá replicar el mismo procedimiento sobre una cuenta que aún no ha sido utilizada: el **Total de Activos No Corrientes**. Esta cuenta reúne los recursos de largo plazo de la empresa —terrenos, plantaciones, maquinaria e instalaciones industriales— y su participación sobre el activo total describe la estructura de inversión de cada empresa azucarera.

**Se solicita:**

1. Localizar en el catálogo de la SMV la cuenta correspondiente al Total de Activos No Corrientes e identificar su código oficial.
2. Recuperar el valor de esa cuenta para las cuatro empresas analizadas.
3. Incorporar la variable a `perfil_reto`.
4. Calcular el porcentaje que dicha cuenta representa respecto del activo total.

**Tiempo estimado:** 10 a 12 minutos.


In [10]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 1
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------


# Revisamos el catálogo para confirmar la cuenta de activos no corrientes.
muestra = balance[balance["nombre_empresa"] == empresas[0]]
columna_desc = resolver_columna(muestra, ["descripcion_cuenta", "descripcion"])

catalogo_no_corriente = muestra[
    muestra[columna_desc].astype(str).str.contains("No Corrientes", case=False, na=False)
][["cuenta", columna_desc, "monto1"]]

print("Empresa de referencia:", empresas[0])
display(catalogo_no_corriente.reset_index(drop=True))

# Código oficial del Total de Activos No Corrientes.
TOTAL_ACTIVO_NO_CORRIENTE = "1D02ST"

print("Código declarado:", TOTAL_ACTIVO_NO_CORRIENTE)

Empresa de referencia: AGROINDUSTRIAL LAREDO S.A.A.


,cuenta,descripcion_cuenta,monto1
0,1D0118,Total Activos Corrientes Distintos de los Acti...,137966.0
1,1D0115,Activos no Corrientes o Grupos de Activos para...,0.0
2,1D0223,"Activos por Impuestos Corrientes, no Corrientes",0.0
3,1D0224,Activos no Corrientes Distintos al Efectivo Pi...,0.0
4,1D02ST,Total Activos No Corrientes,802358.0
5,1D04ST,Total Pasivos No Corrientes,202039.0


Código declarado: 1D02ST


In [11]:
# Recupera el total de activos no corrientes usando su código oficial.
activo_no_corriente = extraer_total_corriente(
    balance,
    TOTAL_ACTIVO_NO_CORRIENTE,
    "activo_no_corriente"
)

# Agrega la nueva cuenta al perfil de trabajo.
perfil_reto = perfil_reto.merge(
    activo_no_corriente,
    on="empresa",
    how="inner"
)

# Calcula qué proporción del activo total corresponde a activos no corrientes.
perfil_reto["pct_activo_no_corriente"] = (
    perfil_reto["activo_no_corriente"] / perfil_reto["activo_total"]
)

# Muestra los resultados.
display(
    perfil_reto[
        ["empresa", "activo_total", "activo_no_corriente", "pct_activo_no_corriente"]
    ].sort_values("pct_activo_no_corriente", ascending=False).round(3)
)

,empresa,activo_total,activo_no_corriente,pct_activo_no_corriente
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1352469.0,1183634.0,0.875
0,AGROINDUSTRIAL LAREDO S.A.A.,940324.0,802358.0,0.853
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1923732.0,1494428.0,0.777
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,926161.0,631228.0,0.682


**Pregunta 4.** ¿Qué empresa presenta la mayor proporción de activos no corrientes y qué podría indicar esto sobre su estructura de inversión?

**Respuesta:**

**EMPRESA AGROINDUSTRIAL POMALCA S.A.A.** registra la proporción más alta de activos no corrientes, con **87.5 %** de sus activos totales. Esto significa que una parte predominante de sus recursos se encuentra concentrada en activos de largo plazo. Como referencia, Laredo presenta **85.3 %**, Casa Grande **77.7 %** y Cartavio **68.2 %**. El resultado refleja una mayor concentración de inversión a largo plazo, pero por sí solo no demuestra una mayor eficiencia.

---


### Ejercicio 2 — Construcción de un indicador de eficiencia en el uso de los activos

Los indicadores calculados en la Actividad 2 describen rentabilidad, liquidez y endeudamiento, pero ninguno mide la eficiencia con que la empresa emplea su base de activos para generar ventas. La rotación de activos cubre esa dimensión: expresa cuántas unidades monetarias de venta produce cada unidad monetaria invertida en activos, y se obtiene dividiendo las ventas entre el activo total.

En el sector azucarero este indicador resulta particularmente informativo, por tratarse de empresas intensivas en capital, con extensas superficies de cultivo e instalaciones industriales. Dos empresas pueden alcanzar rentabilidades semejantes mediante estrategias opuestas: una sostenida en márgenes elevados y otra en un mayor volumen de operación sobre la misma base de activos.

**Se solicita:**

1. Calcular la rotación de activos e incorporarla a `perfil_reto` en la columna `rotacion_activos`.
2. Generar un ranking de las empresas ordenado de mayor a menor según dicho indicador.
3. Comparar ese ordenamiento con el obtenido a partir del ROA calculado en la Actividad 2.

**Tiempo estimado:** 10 a 12 minutos.


In [12]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 2
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------


# Calcula cuántas veces las ventas representan el activo total.
perfil_reto["rotacion_activos"] = (
    perfil_reto["ventas"] / perfil_reto["activo_total"]
)

# Ordena las empresas de mayor a menor rotación.
ranking_rotacion = perfil_reto[
    ["empresa", "rotacion_activos", "roa"]
].sort_values(
    "rotacion_activos",
    ascending=False
)

# Muestra el ranking para compararlo con el ROA.
display(ranking_rotacion.round(3))

,empresa,rotacion_activos,roa
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.578,0.099
0,AGROINDUSTRIAL LAREDO S.A.A.,0.467,0.059
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.331,0.038
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.173,0.007


**Pregunta 5.** ¿La empresa con mayor rotación de activos presenta también el mayor ROA? Explique brevemente a partir de los resultados obtenidos.

**Respuesta (máximo cuatro líneas):**

Sí. **Cartavio** encabeza tanto la rotación de activos, con **0.578**, como el **ROA, con 0.099 (9.9 %)**. Por lo tanto, ocupa el primer puesto en ambos indicadores. La rotación expresa cuánto se vende en relación con los activos, mientras el ROA mide la utilidad neta obtenida respecto a esos mismos activos.

---


### Ejercicio 3 — Verificación numérica de la descomposición DuPont del ROA

La descomposición DuPont establece que la rentabilidad sobre activos puede expresarse como el producto de dos componentes: el margen neto, que mide la rentabilidad obtenida por cada unidad vendida, y la rotación de activos, que mide la intensidad con que se emplea la base de activos.

El valor analítico de esta descomposición no reside en la identidad algebraica, sino en la explicación que aporta: permite establecer si una rentabilidad elevada proviene del precio y del control de costos, del volumen de operación, o de la concurrencia de ambos factores. Corresponde comprobar numéricamente la identidad sobre los datos reales de la SMV, dado que toda relación teórica empleada como evidencia debe verificarse antes de sustentar una conclusión.

**Se solicita:**

1. Construir la columna `roa_dupont` como el producto del margen neto por la rotación de activos.
2. Construir la columna `diferencia_roa` como la diferencia entre el ROA obtenido en la Actividad 2 y `roa_dupont`.
3. Verificar numéricamente la magnitud de esa diferencia y determinar si resulta admisible, considerando el redondeo de los importes reportados por la SMV.

**Tiempo estimado:** 10 a 12 minutos.


In [13]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 3
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------

# Calcula el ROA mediante la descomposición DuPont.
perfil_reto["roa_dupont"] = (
    perfil_reto["margen_neto"] * perfil_reto["rotacion_activos"]
)

# Compara el ROA original con el ROA obtenido mediante DuPont.
perfil_reto["diferencia_roa"] = (
    perfil_reto["roa"] - perfil_reto["roa_dupont"]
)

# Muestra los valores para verificar la diferencia.
display(
    perfil_reto[
        ["empresa", "margen_neto", "rotacion_activos", "roa", "roa_dupont", "diferencia_roa"]
    ].round(6)
)

,empresa,margen_neto,rotacion_activos,roa,roa_dupont,diferencia_roa
0,AGROINDUSTRIAL LAREDO S.A.A.,0.126336,0.466790,0.058972,0.058972,-0.0
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.171651,0.577733,0.099169,0.099169,0.0
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.115246,0.331481,0.038202,0.038202,0.0
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.041172,0.173050,0.007125,0.007125,0.0


> **Interpretación:**
Los resultados confirman que el ROA también puede calcularse como el producto del margen neto y la rotación de activos. La columna `diferencia_roa` sirve para contrastar ese valor con el ROA obtenido previamente; si la diferencia es mínima, puede explicarse por el redondeo aplicado a los importes reportados por la SMV.


**Pregunta 6.** ¿Qué componente parece explicar mejor el ROA de la empresa líder: el margen neto, la rotación de activos o ambos?

**Respuesta (máximo cuatro líneas):**

En **Cartavio**, que lidera el ROA, el resultado se relaciona con **los dos componentes**. La empresa posee el mayor **margen neto (17.2 %)** y la mayor **rotación de activos (0.578)** del grupo. En consecuencia, su **ROA de 9.9 %** reúne tanto una rentabilidad superior sobre las ventas como un mayor nivel de ventas generado por sus activos.

---


### Ejercicio 4 — Consulta empresarial con condiciones múltiples en DuckDB

En la Actividad 2 se empleó DuckDB para reproducir un ordenamiento ya obtenido con Polars, con la finalidad de comprobar la reproducibilidad del resultado entre dos motores distintos. Una consulta de negocio, sin embargo, no se limita a ordenar: delimita un subconjunto de la población mediante criterios definidos previamente por la gerencia.

En este ejercicio deberá construir una consulta que identifique aquellas empresas que satisfacen **simultáneamente** dos condiciones financieras —capacidad de atender obligaciones de corto plazo y generación de rentabilidad positiva para el accionista— y que presente el resultado ordenado según su exposición al endeudamiento. Este es el procedimiento habitual mediante el cual un área de riesgos delimita una cartera de empresas elegibles.

**Se solicita:**

1. Construir sobre `perfil_reto` una consulta SQL que incluya `SELECT`, `WHERE`, condiciones múltiples enlazadas con `AND` y `ORDER BY`.
2. Establecer como criterios una razón corriente mayor o igual a 1 y un ROE positivo.
3. Ordenar el resultado por endeudamiento, de menor a mayor.
4. Determinar cuántas empresas satisfacen ambos criterios.

**Tiempo estimado:** 10 a 12 minutos.


In [14]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 4
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------

# Consulta las empresas que cumplen ambos criterios financieros.
CONSULTA_RETO4 = """
SELECT
    empresa,
    razon_corriente,
    roe,
    endeudamiento
FROM perfil_reto
WHERE razon_corriente >= 1
  AND roe > 0
ORDER BY endeudamiento ASC
"""

# Ejecuta la consulta sobre el perfil de trabajo.
resultado_reto4 = duckdb.sql(CONSULTA_RETO4).df()

# Muestra las empresas que cumplen las condiciones.
display(resultado_reto4.round(3))

# Cuenta cuántas empresas cumplen ambos criterios.
print("Empresas que cumplen ambos criterios:", len(resultado_reto4))

,empresa,razon_corriente,roe,endeudamiento
0,AGROINDUSTRIAL LAREDO S.A.A.,1.101,0.090,0.348
1,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1.184,0.070,0.454
2,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1.554,0.014,0.502


Empresas que cumplen ambos criterios: 3


**Pregunta 7.** Según los criterios establecidos, ¿qué empresa presenta la condición financiera más favorable dentro del grupo analizado?

**Respuesta (máximo cuatro líneas):**

**Agroindustrial Laredo S.A.A.** presenta la condición financiera más favorable bajo los criterios definidos. Registra una razón corriente de **1.101**, mantiene un **ROE positivo de 9.0 %** y, entre las empresas que cumplen ambas condiciones, posee el endeudamiento más bajo, con **34.8 %**.

---


### Ejercicio 5 — Representación conjunta de la rentabilidad y del nivel de endeudamiento

Los gráficos de barras construidos en la Actividad 2 comparan una dimensión por vez. La decisión empresarial, sin embargo, rara vez depende de un solo indicador: una rentabilidad elevada sostenida sobre un endeudamiento igualmente elevado no equivale a una rentabilidad semejante obtenida con una estructura financiera conservadora, aun cuando ambas presenten el mismo ROE.

El gráfico de dispersión permite examinar dos dimensiones de manera simultánea y, mediante el tamaño del marcador, incorporar una tercera. Al situar el endeudamiento en el eje horizontal y el ROE en el eje vertical, cada cuadrante adquiere una lectura gerencial precisa: la combinación de rentabilidad elevada con endeudamiento reducido corresponde al perfil más favorable, mientras que la combinación inversa exige un examen detenido.

**Se solicita:**

1. Construir con Plotly un gráfico de dispersión que sitúe el endeudamiento en el eje horizontal y el ROE en el eje vertical.
2. Identificar cada punto con el nombre de la empresa correspondiente.
3. Representar el activo total mediante el tamaño del marcador.
4. Rotular los ejes y titular el gráfico de modo que resulte interpretable sin recurrir al código.

**Tiempo estimado:** 10 a 12 minutos.


In [15]:
# ---------------------------------------------------------------------------
# CELDA DE TRABAJO — Ejercicio 5
# Desarrolle aquí su código.
# ---------------------------------------------------------------------------


# Dispersión: endeudamiento vs. ROE
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=perfil_reto["endeudamiento"],
    y=perfil_reto["roe"],
    mode="markers+text",
    text=perfil_reto["empresa"],
    textposition="top center",
    marker=dict(
        size=perfil_reto["activo_total"] / 20000,
        sizemode="diameter"
    ),
    hovertemplate=(
        "<b>%{text}</b><br>"
        "Endeudamiento: %{x:.1%}<br>"
        "ROE: %{y:.1%}<extra></extra>"
    )
))

fig.update_layout(
    title="Rentabilidad (ROE) y nivel de endeudamiento",
    xaxis_title="Endeudamiento",
    yaxis_title="ROE",
    template="plotly_white"
)

fig.show()

> **Interpretación**: La visualización permite observar al mismo tiempo la rentabilidad y el nivel de endeudamiento de cada compañía. De acuerdo con los criterios del ejercicio, una empresa con mayor ROE y menor endeudamiento presenta una combinación relativamente más favorable.


**Pregunta 8.** ¿Qué empresa muestra la combinación más favorable entre rentabilidad y nivel de endeudamiento?

**Respuesta (máximo cuatro líneas):**

**Agroindustrial Laredo S.A.A.** presenta la combinación más favorable de acuerdo con los criterios planteados. Su **ROE es de 9.0 %** y su endeudamiento, de **34.8 %**, es el menor del grupo. Así, mantiene rentabilidad positiva con una estructura financiera relativamente menos expuesta.

---


### Retroalimentación final — Vinculación con el modelo DELTA Plus

Los cinco ejercicios anteriores se resolvieron mediante Python, DuckDB y Plotly sobre datos financieros oficiales. Corresponde ahora situar ese trabajo dentro del marco conceptual de la sesión. Responda de manera breve.

> **Pregunta 9.** A partir del análisis realizado mediante Python, identifique qué dimensión de DELTA Plus se evidencia directamente en esta actividad: *Technology*, *Analytical Techniques* o ambas. Justifique su respuesta en un máximo de tres líneas.

**Respuesta:**
Se observan **Technology** y **Analytical Techniques**, ya que Python, DuckDB y Plotly se emplearon para procesar, consultar y representar los datos financieros. Además, se aplicaron procedimientos descriptivos y comparativos mediante diferentes indicadores.

---

> **Pregunta 10.** *Technology.* ¿Qué tecnologías resultarían necesarias para que una de las empresas analizadas pudiera ejecutar de manera periódica y reproducible el análisis desarrollado en este laboratorio?

**Respuesta (máximo cuatro líneas):**
Se requerirían herramientas como **Python**, una base de datos y una plataforma de visualización conectadas con fuentes de información actualizadas. También sería conveniente automatizar la extracción, transformación y generación periódica de indicadores y gráficos para asegurar la reproducibilidad.

---

> **Pregunta 11.** *Analytical Techniques.* ¿En qué nivel de técnicas analíticas se sitúa el análisis realizado en esta sesión y qué técnica adicional podría incorporar la organización para mejorar sus decisiones?

**Respuesta (máximo cuatro líneas):**
El trabajo corresponde principalmente a los niveles **descriptivo y diagnóstico**, porque calcula indicadores y compara el desempeño financiero de las empresas. Como siguiente paso, podría incorporarse **analítica predictiva** para proyectar escenarios y brindar mayor anticipación a la toma de decisiones.

---


## Conclusiones

Registre tres conclusiones técnicas y de negocio:

1. Los indicadores calculados permitieron contrastar a las empresas desde las dimensiones de rentabilidad, liquidez y endeudamiento, mostrando diferencias claras en sus respectivos perfiles financieros.
2. De acuerdo con los criterios establecidos en el laboratorio, **Agroindustrial Laredo S.A.A.** reunió la combinación más favorable, con un **ROE de 9.0 %**, una razón corriente de **1.101** y el endeudamiento más bajo, de **34.8 %**.
3. El uso conjunto de **Python, DuckDB y Plotly** hizo posible desarrollar un análisis reproducible desde varias dimensiones; sin embargo, los datos financieros por sí solos no son suficientes para establecer la madurez analítica de las empresas mediante el modelo DELTA Plus.


## Referencias oficiales

- Superintendencia del Mercado de Valores. Portal de Datos Abiertos y servicio de información financiera.
- Banco Central de Reserva del Perú. Series estadísticas mensuales.
